src/api/polkadot/connectApi.ts

```diff
@@ -1,8 +1,7 @@
-import { ApiPromise } from '@polkadot/api';
+import { ApiPromise, WsProvider } from '@polkadot/api';
 import { keyring } from '@polkadot/ui-keyring';
 import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
 import { isTestChain } from '@polkadot/util';
-import { WsProvider } from '@polkadot/rpc-provider';
 import * as plasmDefinitions from '@plasm/types/interfaces/definitions';
```

Comment:
You can fix the `instantiateWithCode(endowment, gas_limit, code, data, salt)` error by updating the `@plasm/types` to `^1.5.4-2`. Here is how you can use the new import

```ts

import chainTypeDefs from '@plasm/types';
...

export const connectApi = async (endpoint: string) => {
  const provider = new WsProvider(endpoint);
  const api = new ApiPromise({
    provider,
    types: {
      ...chainTypeDefs.dustyDefinitions,
      LookupSource: 'MultiAddress',
    },
  });

  try {
    await api.isReadyOrError;
  } catch (err) {
    console.error('err', err);
  }

  try {
    await loadAccounts(api);
  } catch (err) {
    console.error(err);
  }

  // load the web3 extension
  injectedPromise
    .then((): void => {})
    .catch((error: Error) => console.error(error));

  return api;
};
```